backend/hdf5/h5x/H5Group.cpp

```diff
@@ -217,8 +218,12 @@ DataSet H5Group::createData(const std::string &name,
         HErr res = H5Pset_chunk(dcpl.h5id(), rank, chunks.data());
         res.check("Could not set chunk size on data set creation plist");
     }
-
-    DataSet ds = H5Dcreate(hid, name.c_str(), fileType.h5id(), space.h5id(), H5P_DEFAULT, dcpl.h5id(), H5P_DEFAULT);
+    DataSet ds;
+    if (compression) {
+         HErr status = H5Pset_deflate (dcpl.h5id(), 6);
+         status.check("Could not set compression!");
```

Comment:
Which exception will this result in? Maybe we want to differentiate this use case client-side, so we a program could try to use compression, and if fails retry without it?